native/src/main/java/io/ballerina/stdlib/http/api/HttpResponseInterceptorUnitCallback.java

```diff
@@ -63,6 +66,9 @@ public void notifySuccess(Object result) {
             invokeErrorInterceptors((BError) result, true);
             return;
         }
+        if (possibleLastInterceptor) {
```

Comment:
What is not `lastInterceptor` but `possibleLastInterceptor `?

Reply:
Since we are not aware of the logic inside the last response interceptor in the pipeline, it is not possible to determine the last interceptor (It is possible to return an error from the last response interceptor, then the last interceptor should be a response error interceptor). That's why I mark the interceptors as `possibleLastInterceptors` and clean the request message only when the return type is not `error` otherwise the request content will be released in the next response error interceptors.